patroni/postgresql/available_parameters/0_postgres.yml

```diff
@@ -0,0 +1,2132 @@
+parameters:
+  allow_in_place_tablespaces:
+  - type: Bool
+    version_from: 150000
+    version_till: null
```

Comment:
These classes used to be namedetuples, where position of the argument in constructor plays a big role.
Now we can always use named parameters and make some of them (`version_till`, `unit`) default to `None`.
This way we can reduce a bit the size of the YAML file.

Reply:
Oh, good point! I'll work on that in a few to reduce the size of that file.
When I initially generated this YAML I had not yet pulled the merged changes from your pyright PR, so I still had the namedtuple definitions. With that in mind at that point I just generated the YAML file with a Python script using the `_fields` attribute and the class names.